source/core/src/main/com/csse3200/game/screens/AnimalSelectionScreen.java

```diff
@@ -0,0 +1,183 @@
+package com.csse3200.game.screens;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.ScreenAdapter;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Texture;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.InputEvent;
+import com.badlogic.gdx.scenes.scene2d.Stage;
+import com.badlogic.gdx.scenes.scene2d.ui.Dialog;
+import com.badlogic.gdx.scenes.scene2d.ui.Skin;
+import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
+import com.badlogic.gdx.scenes.scene2d.ui.Image;
+import com.badlogic.gdx.scenes.scene2d.ui.Table;
+import com.badlogic.gdx.scenes.scene2d.utils.ChangeListener;
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+import com.csse3200.game.GdxGame;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AnimalSelectionScreen extends ScreenAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(AnimalSelectionScreen.class);
+    private final GdxGame game;
+    private Stage stage;
+    private Table mainTable;
+    private Image selectedAnimalImage;
+    private TextButton selectButton;
+
+    public AnimalSelectionScreen(GdxGame game) {
+        this.game = game;
```

Comment:
Should this function be registering services here? Ie the input service to handle button clicks (I'm not sure if you should use that or the way you have done it is better - someone else can check that).